test/functional/test_runner.py

```diff
@@ -294,6 +295,9 @@ def main():
             if not exclude_list:
                 print("{}WARNING!{} Test '{}' not found in current test list.".format(BOLD[1], BOLD[0], exclude_test))
 
+    if args.filter:
+        test_list = list(filter(re.compile(args.filter).search, test_list))
```

Comment:
Tested successfully with one exception: passing "tool" as in `test_runner.py --filter tool` or `test/functional/test_runner.py --filter tool` does no filtering and runs all the tests.